Fix printing both index signatures without type annotations and multi-parameter index signatures

cola119 opened the pull request:
fixes https://github.com/prettier/prettier/issues/7225
- [x] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory)
- [x] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/pr-XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

cola119 commented:
I guess I can fix https://github.com/prettier/prettier/issues/7224 quickly, but should I commit and push to this branch to avoid conflicts? Or is it better to create a new PR for that?

thorn0 commented on changelog_unreleased/typescript/pr-7228.md:
```diff
@@ -0,0 +1,11 @@
+#### Typescript: Remove unnecessary colons for TSIndexSignature without type parameters ([#7228](https://github.com/prettier/prettier/pull/7228) by [@cola119](https://github.com/cola119))
```
```suggestion
#### Fix printing index signatures without type annotations ([#7228](https://github.com/prettier/prettier/pull/7228) by [@cola119](https://github.com/cola119))
```

thorn0 commented on tests/typescript_index_signature/__snapshots__/jsfmt.spec.js.snap:
```diff
@@ -11,11 +11,14 @@ class LocalStorage implements Storage {
   [key: string]: any;
 }
 
+type A = { [key: string] };
```
As an index signature without a type annotation is not valid TS, please move the test for this case to a separate file in `tests/typescript_error_recovery`.

thorn0 commented:
@cola119 I'd say it's okay to fix #7224 in this PR.

cola119 commented:
@thorn0 thanks your reviews. I modified the pointed out items and fixed https://github.com/prettier/prettier/issues/7224. 
Sorry in my poor English but please give me a review :bow:

cola119 commented:
@sosukesuzuki san,
I fixed to support long expressions and trailing comma. Please let me know if there are any problems.
